studiopress/pattern-manager
# Change order of deletion when renaming to fix asset handling bug

## Tasks


### Summary of changes
This PR fixes a bug with asset handling when renaming a pattern. 

When a pattern with images is renamed, its pattern file **and its image files are deleted FIRST**, and then the new pattern file is created and scanned for images. Because those image URLs are already referencing the current theme, they [are not found](https://github.com/studiopress/pattern-manager/blob/95d56e11f6b64dc8baa4fc5bfa2b945b1289e50d/wp-modules/pattern-data-handlers/pattern-data-handlers.php#L486), and thus don't get replaced with a PHP tag, or moved back into the theme.

This can be fixed by deleting the original pattern **after the new one has been created**.

### How to test the bug on `main`
1. Make a pattern with an image
2. Make another pattern with a different image
3. Rename the second pattern.
4. Notice the reference to the image in the file no longer uses the PHP tag, but hardcodes the URL to the image, and the image has been removed from your theme.

### How to test the fix on `fix/aset-handling-after-rename`
1. Make a pattern with an image
2. Make another pattern with a different image
3. Rename the second pattern.
4. Notice the reference to the image in the file **still uses the PHP tag**, and the image has NOT been removed from your theme.
